src/FiniteElement.m

```diff
@@ -1037,19 +1037,115 @@
+                            sigma_f_c_B = obj.prev_fe.w(obj.prev_fe.ind_Y_gap{1})+sum2(horzcat(f_c_B{:}));
+                        end
+                        sigma_l_n_F = sum2(horzcat(l_n_F{:}));
+                        sigma_l_n_B = sum2(horzcat(l_n_B{:}));
+
+                        pi_f_c = obj.w(obj.ind_Y_gap{1}).*sigma_f_c_F.*sigma_f_c_B;
+                        pi_l_n = sigma_l_n_F .* sigma_l_n_B;
+                        kappa = pi_f_c + pi_l_n;
+                        if obj.model.friction_exists
+                            % Go one by one for each contact (TODO vectorize)
+                            zeta = SX.ones(obj.model.dims.n_contacts, 1);
+                            for ii=1:obj.model.dims.n_contacts
+                                if obj.model.mu_f(ii)
+                                    if strcmp(obj.problem_options.friction_model, 'Conic')
+                                        xi_p_F = cellfun(@(x) obj.w(x), obj.ind_p_vt, 'uni', false);
+                                        xi_p_B = cellfun(@(x) obj.prev_fe.w(x), obj.prev_fe.ind_p_vt, 'uni', false);
+                                        xi_n_F = cellfun(@(x) obj.w(x), obj.ind_n_vt, 'uni', false);
+                                        xi_n_B = cellfun(@(x) obj.prev_fe.w(x), obj.prev_fe.ind_n_vt, 'uni', false);
+                                        beta_F = cellfun(@(x) obj.w(x), obj.ind_beta_conic, 'uni', false);
+                                        beta_B = cellfun(@(x) obj.prev_fe.w(x), obj.prev_fe.ind_beta_conic, 'uni', false);
+
+                                        sigma_xi_p_F = obj.w(obj.ind_P_vt{1})+sum2(horzcat(xi_p_F{:}));
+                                        sigma_xi_n_F = obj.w(obj.ind_N_vt{1})+sum2(horzcat(xi_n_F{:}));
+                                        if isempty(obj.prev_fe.ind_P_vt{1})
+                                            sigma_xi_p_B = sum2(horzcat(xi_p_B{:}));
+                                            sigma_xi_n_B = sum2(horzcat(xi_n_B{:}));
+                                        else
+                                            sigma_xi_p_B = obj.prev_fe.w(obj.prev_fe.ind_P_vt{1})+sum2(horzcat(xi_p_B{:}));
+                                            sigma_xi_n_B = obj.prev_fe.w(obj.prev_fe.ind_N_vt{1})+sum2(horzcat(xi_n_B{:}));
+                                        end
+                                        sigma_beta_F = sum2(horzcat(beta_F{:}));
+                                        sigma_beta_B = sum2(horzcat(beta_B{:}));
+
+                                        pi_xi_p = sigma_xi_p_F .* sigma_xi_p_B;
+                                        pi_xi_n = sigma_xi_n_F .* sigma_xi_n_B;
+                                        pi_beta = sigma_beta_F .* sigma_beta_B;
+
+                                        zeta(ii) = (sigma_f_c_B + sigma_f_c_F) + (sum(pi_xi_p + pi_xi_n) + pi_beta);
+                                    else
+                                        % TODO: Verify
```

Comment:
I may be missing things in the polyhedral implementation. Need to think whether we need the left boundary points here for switches that switch while slipping (only $\gamma_{n,0} = 0$?).